### PR TITLE
[added] add class to html when open the modal.

### PR DESCRIPTION
**Pull request by diasbruno**
This is a continuation and review of #588 (feature by @bannier).

Changes proposed:

- add class to `html` tag when modal is open.
- this feature will follow the same rules as `document.body` class.

Acceptance Checklist:
- [x] All commits have been squashed to one.
- [x] The commit message follows the guidelines in `CONTRIBUTING.md`.
- [x] Documentation (README.md) and examples have been updated as needed.
- [x] If this is a code change, a spec testing the functionality has been added.
- [x] If the commit message has [changed] or [removed], there is an upgrade path above.

### CODE REVIEW

**Comment by diasbruno** [comment]
@ianprime0509 @bannier Can you help me review this PR? I just made the `htmlOpenClassName` to follow the same rules as `bodyOpenClassName`.

File: examples/basic/app.css [hunk 1]
```diff
@@ -29,3 +29,8 @@
   transform: scale(0.5) rotateX(30deg);
   transition: all 150ms ease-in;
 }
+
+.ReactModal__Body--open,
+.ReactModal__Html--open {
```
Comment (ianprime0509) [inline examples/basic/app.css]: It looks like the `ReactModal__Html--open` class is never being applied to the HTML element (unless it is set explicitly); in fact, this is tested on line 279 of `specs/Modal.spec.js`.  On the other hand, the `ReactModal__Body--open` class is applied to the body by default, which seems potentially inconsistent.

File: specs/Modal.spec.js [hunk 2]
```diff
@@ -253,18 +255,33 @@ export default () => {
...
+    isHtmlWithReactModalOpenClass().should.not.be.ok();
+  });
+
+  it("append class to document.body if modal is open.", () => {
     renderModal({ isOpen: true });
     isBodyWithReactModalOpenClass().should.be.ok();
-    unmountModal();
   });
 
+  it("don't append class to <html /> if not defined.", () => {
```
Comment (ianprime0509) [inline specs/Modal.spec.js]: Here's the test that I mentioned above about the HTML class not being applied by default.

File: examples/basic/app.css (hunk 1 above)
Comment (diasbruno) [reply examples/basic/app.css]: At least one example must use this property, I guess.

**Comment by diasbruno** [comment]
Thanks for pointing out the docs. I'll update it.

**Comment by diasbruno** [comment]
I think everything should be ok now. I'll leave the example to other PR.

File: docs/README.md [hunk 3]
```diff
@@ -69,6 +69,11 @@ import ReactModal from 'react-modal';
      See the `Styles` section for more details.
   */
   bodyOpenClassName="ReactModal__Body--open"
+  /*
+     String className to be applied to the document.html (must be a constant string).
+     See the `Styles` section for more details.
```
Comment (diasbruno) [inline docs/README.md]: Should explicit tell that this attribute is `null` by default.

Comment (diasbruno) [reply docs/README.md]: Done.

File: specs/Modal.spec.js (hunk 2 above)
Comment (ianprime0509) [reply specs/Modal.spec.js]: Yes, that makes sense, thanks for clarifying!

Comment (diasbruno) [reply specs/Modal.spec.js]: It's more of a conservative approach, but we can later change to define it by default.

**Comment by diasbruno** [comment]
Thanks @ianprime0509 and @bannier for helping with this. 👍

File: specs/Modal.spec.js (hunk 2 above)
Comment (woniesong92) [reply specs/Modal.spec.js]: Hey @diasbruno, shouldn't doc mention that you have to explicitly define `htmlOpenClassName` for it to work?

Comment (diasbruno) [reply specs/Modal.spec.js]: @woniesong92 I updated the docs to mention that this prop is `null` by default. I'll check if the published doc is not updated, but I'm sure I publish `3.3.1` and the new docs.

File: docs/README.md [hunk 4]
```diff
@@ -69,6 +69,12 @@ import ReactModal from 'react-modal';
      See the `Styles` section for more details.
   */
   bodyOpenClassName="ReactModal__Body--open"
+  /*
+     String className to be applied to the document.html (must be a constant string).
+     This attribute is `null` by default.
+     See the `Styles` section for more details.
+  */
+  htmlOpenClassName="ReactModal__Html--open"
```
Comment (diasbruno) [inline docs/README.md]: @woniesong92 Here.

File: docs/styles/classes.md [hunk 5]
```diff
@@ -61,6 +61,21 @@ non-default `bodyOpenClassName`), you could use the following CSS:
 }
 ```
 
+You can define a class to be added to the html tag, using the `htmlOpenClassName`
+attribute, which can be helpeful to stop the page to scroll to the top when open
+a modal.
```
Comment (diasbruno) [inline docs/styles/classes.md]: @woniesong92 Ah...I think I could add something here to remind that it's `null`.

**Comment by jaspersorrio** [comment]
Hello I tried using this feature today but couldn't even after npm install -save react-modal@latest.

Is the code on npm updated?

**Comment by diasbruno** [comment]
Hi @jaspersorrio, can you open a issue for this, please?

**Comment by ElixirMike** [comment]
Trying to follow this thread. So, what exactly do I need to do in order to not have the scroll to the top of the window when openiing the modal? It's not clear what I need to add where?

**Comment by nolandg** [comment]
Has the scrolling to top on open issue ever been solved? I've tried overflow, position, tab index, everything I could find in these threadsy but nothing has worked.

**Pull request merged**